methods/incidentReports.coffee

```diff
@@ -21,15 +21,22 @@ Meteor.methods
     return newId
 
   # similar to editIncidentReport, but allows you to set a single field without changing any other existing fields.
-  updateIncidentReport: (incident) ->
+  updateIncidentReport: (incident, fieldsToRemove=[]) ->
     _id = incident._id
     delete incident._id
     user = Meteor.user()
     if not Roles.userIsInRole(user._id, ['admin'])
       throw new Meteor.Error("auth", "User does not have permission to edit incidents")
     incident.modifiedByUserId = user._id
     incident.modifiedByUserName = user.profile.name
-    res = Incidents.update({_id: _id}, {$set: incident})
+    updateOperators =
+      $set: incident
+    if fieldsToRemove.length
+      fieldsToUnset = {}
+      fieldsToRemove.forEach (field) ->
```

Comment:
The incident obj we pass to the method is a combination of the old and new (user entered via the form). If the type is changed it will have both properties. So would you recommend querying for the old and doing the a comparison in the method?